chicago/views.py

```diff
@@ -338,6 +345,81 @@ class ChicagoCommitteeDetailView(CommitteeDetailView):
 class ChicagoEventsView(EventsView):
     template_name = "events.html"
 
+    def get_context_data(self, **kwargs):
+        context = super(EventsView, self).get_context_data(**kwargs)
+
+        # Get year range for datepicker.
+        aggregates = ChicagoEvent.objects.aggregate(
+            Min("start_time"), Max("start_time")
+        )
+
+        context["year_range_min"] = aggregates["start_time__min"].year
+        context["year_range_max"] = aggregates["start_time__max"].year
+
+        # Did the user set date boundaries?
+        date_str = self.request.GET.get("form_datetime")
+        day_grouper = lambda x: x.local_start_time.date  # noqa E731
+        context["select_date"] = ""
+
+        date_time = timezone.now()
+        # If yes, then filter for dates.
+        if date_str:
+            context["date"] = date_str
+            date_time = parser.parse(date_str)
+
+            select_events = (
+                ChicagoEvent.objects.filter(start_time__gt=date_time)
+                .filter(start_time__lt=(date_time + relativedelta(months=1)))
+                .order_by("start_time")
+            )
+
+            org_select_events = []
+
+            for event_date, events in itertools.groupby(select_events, key=day_grouper):
```

Comment:
it would be very nice to push this grouping logic into the templates with the regroup tag.

https://openfolder.sh/grouping-in-django-templates

again this is not a required change, we can save it for later.

Reply:
i'll open up another issue to do a refactor of the events page